packages/data-point/lib/entity-types/allowed-modifiers/allowed-modifiers.js

```diff
@@ -0,0 +1,38 @@
+const Util = require('util')
+const _ = require('lodash')
+
+function allowedProperties (id, spec, validKeys) {
+  const differentKeys = _.difference(Object.keys(spec), validKeys)
+  if (differentKeys.length > 0) {
+    throw new Error(
+      Util.format(
+        'Entity "%s" did not recognize the following properties: %s, valid properties are: %s.\nPlease review your entity and make any corrections so it can be parsed:\n\'%s\': %s',
+        id,
+        differentKeys.join(', '),
+        validKeys.join(', '),
+        id,
+        Util.inspect(spec)
+      )
+    )
+  }
+
+  return true
+}
+
+module.exports.allowedProperties = allowedProperties
+
+const baseModifiers = [
+  'inputType',
+  'before',
+  'value',
+  'after',
+  'outputType',
+  'error',
+  'params'
+]
+function allowedModifiers (id, spec, validKeys) {
+  const modifierKeys = baseModifiers.concat(validKeys)
+  return allowedProperties(id, spec, modifierKeys)
+}
+
+module.exports.allowedModifiers = allowedModifiers
```

Comment:
can we rename `allowedProperties` and `allowedModifiers` so  it's more clear what they're doing?

`verifyProperties`?

`validateProperties`?

`throwForInvalidProperties`?

not sure what, just feel there should be a verb in there

Reply:
yes! good suggestion

Reply:
done